### PR TITLE
Fix for GetMemberWithSameMetadataDefinitionAs

### LINKED ISSUE #67533: GetMemberWithSameMetadataDefinitionAs returns incorrect declaring type
```
  [FAIL] System.Reflection.Tests.TypeInfoTests.GetMemberWithSameMetadataDefinitionAs(openGenericType: typeof(System.Reflection.Tests.TI_GenericTypeWithAllMembers<>), closedGenericType: typeof(System.Reflection.Tests.TI_TypeDerivedFromGenericTypeWithAllMembers<int>), checkDeclaringType: False)
  System.ArgumentException : A MemberInfo that matches 'add_PrivateStaticEvent' could not be found. (Parameter 'member')
     at System.Reflection.Runtime.TypeInfos.RuntimeTypeInfo.GetMemberWithSameMetadataDefinitionAs(MemberInfo) + 0x154
     at System.Reflection.Tests.TypeInfoTests.GetMemberWithSameMetadataDefinitionAs(Type, Type, Boolean) + 0x70
     at System.Reflection.Tests!<BaseAddress>+0x6812ca
     at System.InvokeUtils.CallDynamicInvokeMethod(Object, IntPtr, IntPtr, IntPtr, Object, Object[], BinderBundle, Boolean, Boolean) + 0x150
  [FAIL] System.Reflection.Tests.TypeInfoTests.GetMemberWithSameMetadataDefinitionAs(openGenericType: typeof(System.Reflection.Tests.TI_GenericTypeWithAllMembers<>), closedGenericType: typeof(System.Reflection.Tests.TI_TypeDerivedFromGenericTypeWithAllMembersClosed), checkDeclaringType: False)
  System.ArgumentException : A MemberInfo that matches 'add_PrivateStaticEvent' could not be found. (Parameter 'member')
     at System.Reflection.Runtime.TypeInfos.RuntimeTypeInfo.GetMemberWithSameMetadataDefinitionAs(MemberInfo) + 0x154
     at System.Reflection.Tests.TypeInfoTests.GetMemberWithSameMetadataDefinitionAs(Type, Type, Boolean) + 0x70
     at System.Reflection.Tests!<BaseAddress>+0x6812ca
     at System.InvokeUtils.CallDynamicInvokeMethod(Object, IntPtr, IntPtr, IntPtr, Object, Object[], BinderBundle, Boolean, Boolean) + 0x150
  [FAIL] System.Reflection.Tests.TypeInfoTests.GetMemberWithSameMetadataDefinitionAs(openGenericType: typeof(System.Reflection.Tests.TI_GenericTypeWithAllMembers<>), closedGenericType: typeof(System.Reflection.Tests.TI_TypeDerivedFromGenericTypeWithAllMembers<System.Reflection.Tests.TypeInfoTests+ClassWithMultipleConstructors>), checkDeclaringType: False)
  System.ArgumentException : A MemberInfo that matches 'add_PrivateStaticEvent' could not be found. (Parameter 'member')
     at System.Reflection.Runtime.TypeInfos.RuntimeTypeInfo.GetMemberWithSameMetadataDefinitionAs(MemberInfo) + 0x154
     at System.Reflection.Tests.TypeInfoTests.GetMemberWithSameMetadataDefinitionAs(Type, Type, Boolean) + 0x70
     at System.Reflection.Tests!<BaseAddress>+0x6812ca
     at System.InvokeUtils.CallDynamicInvokeMethod(Object, IntPtr, IntPtr, IntPtr, Object, Object[], BinderBundle, Boolean, Boolean) + 0x150
  [FAIL] System.Reflection.Tests.TypeInfoTests.GetMemberWithSameMetadataDefinitionAs(openGenericType: typeof(System.Reflection.Tests.TI_GenericTypeWithAllMembers), closedGenericType: typeof(System.Reflection.Tests.TI_GenericTypeWithAllMembers), checkDeclaringType: True)
  'add_PrivateStaticEvent' doesn't have the right DeclaringType
  Expected: True
  Actual:   False
     at System.Reflection.Tests.TypeInfoTests.GetMemberWithSameMetadataDefinitionAs(Type, Type, Boolean) + 0x199
     at System.Reflection.Tests!<BaseAddress>+0x6812ca
     at System.InvokeUtils.CallDynamicInvokeMethod(Object, IntPtr, IntPtr, IntPtr, Object, Object[], BinderBundle, Boolean, Boolean) + 0x150
  [FAIL] System.Reflection.Tests.TypeInfoTests.GetMemberWithSameMetadataDefinitionAs(openGenericType: typeof(System.Reflection.Tests.TI_GenericTypeWithAllMembers<>), closedGenericType: typeof(System.Reflection.Tests.TI_GenericTypeWithAllMembers), checkDeclaringType: True)
  'add_PrivateStaticEvent' doesn't have the right DeclaringType
  Expected: True
  Actual:   False
     at System.Reflection.Tests.TypeInfoTests.GetMemberWithSameMetadataDefinitionAs(Type, Type, Boolean) + 0x199
     at System.Reflection.Tests!<BaseAddress>+0x6812ca
     at System.InvokeUtils.CallDynamicInvokeMethod(Object, IntPtr, IntPtr, IntPtr, Object, Object[], BinderBundle, Boolean, Boolean) + 0x150

```

### PR DESCRIPTION
**Pull request by LakshanF**
Fixes #67533.

### CODE REVIEW

File: eng/pipelines/runtime.yml
Comment (jkotas) [inline eng/pipelines/runtime.yml:188]: Is this intentional change?

Comment (jkotas) [reply eng/pipelines/runtime.yml:188]: For CI builds, we want to maximize variety in the subset of configurations that are being tested. This is reducing the variety.

Comment (LakshanF) [reply eng/pipelines/runtime.yml:188]: Yes, the Linux arm64 container was timing out and I wanted to check with windows. @MichalStrehovsky has mentioned an issue impacting running multiple library test projects (I think but not sure) that should be fixed in 7.0 and I wanted to see if this will work

Comment (jkotas) [reply eng/pipelines/runtime.yml:188]: Is this a temporary change that you are going to revert before merging the PR, or is your intention to merge this change?

Also, notice that the runtime jobs did not run at all and no NativeAOT testing was done in the latest CI run. Not sure what caused it.

Comment (LakshanF) [reply eng/pipelines/runtime.yml:188]: This was to be a diagnostic effort and then to decide if this needs to be permanent. Yes, this is blocked by the lack of full test pass and not sure what is causing that.

Comment (MichalStrehovsky) [reply eng/pipelines/runtime.yml:188]: CoreCLR tests don't work on arm64. We're still running those tests on the build machine and the build machine is x64.

Only libs tests can currently run on arm64 because we send those to helix to run.

Comment (LakshanF) [reply eng/pipelines/runtime.yml:188]: The most recent test run has completed successfully but I'll revert to the previous position to double check - the earlier runs kept on failing on arm64 for lib tests.

Comment (LakshanF) [reply eng/pipelines/runtime.yml]: All the checks pass with the original runtime pipeline now

**Review by jkotas — APPROVED** [review]
Thanks!

**Pull request merged**
